prefect_kubernetes/utilities.py

```diff
@@ -0,0 +1,73 @@
+""" Utilities for working with the Python Kubernetes API. """
+from pathlib import Path
+from typing import TypeVar, Union
+
+from kubernetes.client import models as k8s_models
+from prefect.infrastructure.kubernetes import KubernetesJob, KubernetesManifest
+
+base_types = {"str", "int", "float", "bool", "list[str]", "dict(str, str)", "object"}
+
+V1KubernetesModel = TypeVar("V1KubernetesModel")
+
+
+def convert_manifest_to_model(
+    manifest: Union[Path, KubernetesManifest], v1_model_name: str
+) -> V1KubernetesModel:
+    """Recursively converts a `dict` representation of a Kubernetes resource to the
+    corresponding Python model containing the Python models that compose it,
+    according to the `openapi_types` on the class retrieved with `v1_model_name`.
+
+    If `manifest` is a path-like object with a `.yaml` or `.yml` extension, it will be
+    treated as a path to a Kubernetes resource manifest and loaded into a `dict`.
+
+    Args:
+        manifest: A path to a Kubernetes resource manifest or its `dict` representation.
```

Comment:
Not `dict` I think.

Reply:
it does accept / convert a `dict` to model, the edge case where a str / path is passed, its converted by `KubernetesJob.job_from_file` to a `dict` before recursing